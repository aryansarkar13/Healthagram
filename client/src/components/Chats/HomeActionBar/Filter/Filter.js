import React, { Component } from "react";
import { connect } from "react-redux";
import { updateGroupFilterCountry } from "store/actions/group/group";
import Select from "react-select";

export class Filter extends Component {
  state = {
    //This is used for changing the value of select
    selectedOption: null
  };

  renderOptions = () => {
    const defaultOption = { value: "", label: "Filter By Country" };

    let options = [];
    const filterCountry = this.props.filterCountry;
    if (this.props.countries) {
      options.push(defaultOption);
      this.props.countries.map((country, index) => {
        //country._id is the name of the country
        //sorry for the confusing name
        options.push({
          value: country._id,
          label: country._id
        });
      });
    }
    return (
      <Select
        options={options}
        value={
          this.state.selectedOption === null
            ? defaultOption
            : this.state.selectedOption
        }
        onChange={this.onSelectChange}
      />
    );
  };

  onSelectChange = selectedOption => {
    console.log("select change is called", selectedOption.value);
    this.setState({ selectedOption });
    this.props.updateGroupFilterCountry(selectedOption.value);
  };

  render() {
    return <div className="Filter">{this.renderOptions()}</div>;
  }
}

const mapStateToProps = state => ({
  countries: state.group.countries,
  filterCountry: state.group.filter.country
});

export default connect(
  mapStateToProps,
  { updateGroupFilterCountry }
)(Filter);
