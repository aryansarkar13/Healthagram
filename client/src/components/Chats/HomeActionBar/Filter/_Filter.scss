.Filter {
  width: 50%;

  @media (max-width: $sidenav-disappear-width) {
    width: 100%;
    margin-top: 10px;
    margin-bottom: 10px;
  }
}
