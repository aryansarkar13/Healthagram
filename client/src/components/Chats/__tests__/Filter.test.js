import { Filter } from "components/Home/Filter/Filter";
import React from "react";
import { shallow } from "enzyme";

describe("The Filter component", () => {
  it("should not regress", () => {
    const countries = [
      {
        _id: "Nepal"
      },
      {
        _id: "Singapore"
      }
    ];
    const wrapper = shallow(<Filter countries={countries} />);
    expect(wrapper).toMatchSnapshot();
  });
});
