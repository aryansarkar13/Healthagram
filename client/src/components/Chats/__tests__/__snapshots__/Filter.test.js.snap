// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Filter component should not regress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    countries={
        Array [
            Object {
              "_id": "Nepal",
            },
            Object {
              "_id": "Singapore",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <select>
          <option
                    selected={true}
                    value="country"
          >
                    Select By Country
          </option>
          <option
                    value="Nepal"
          >
                    Nepal
          </option>
          <option
                    value="Singapore"
          >
                    Singapore
          </option>
</select>,
        <button>
          Apply
</button>,
      ],
      "className": "Filter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              selected={true}
              value="country"
>
              Select By Country
</option>,
            Array [
              <option
                value="Nepal"
>
                Nepal
</option>,
              <option
                value="Singapore"
>
                Singapore
</option>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select By Country",
              "selected": true,
              "value": "country",
            },
            "ref": null,
            "rendered": "Select By Country",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "Nepal",
              "value": "Nepal",
            },
            "ref": null,
            "rendered": "Nepal",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Singapore",
              "value": "Singapore",
            },
            "ref": null,
            "rendered": "Singapore",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Apply",
        },
        "ref": null,
        "rendered": "Apply",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select>
            <option
                        selected={true}
                        value="country"
            >
                        Select By Country
            </option>
            <option
                        value="Nepal"
            >
                        Nepal
            </option>
            <option
                        value="Singapore"
            >
                        Singapore
            </option>
</select>,
          <button>
            Apply
</button>,
        ],
        "className": "Filter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                selected={true}
                value="country"
>
                Select By Country
</option>,
              Array [
                <option
                  value="Nepal"
>
                  Nepal
</option>,
                <option
                  value="Singapore"
>
                  Singapore
</option>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select By Country",
                "selected": true,
                "value": "country",
              },
              "ref": null,
              "rendered": "Select By Country",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "Nepal",
                "value": "Nepal",
              },
              "ref": null,
              "rendered": "Nepal",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "Singapore",
                "value": "Singapore",
              },
              "ref": null,
              "rendered": "Singapore",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Apply",
          },
          "ref": null,
          "rendered": "Apply",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
